### PR TITLE
Used LabelEncoder to convert binary labels in [1, 0] in SearchLight

### LINKED ISSUE #3697: BUG: _SearchLight when 'scoring=roc_auc' & y not in (0, 1)
@kaichogami we missed a necessary feature when adding the `scoring` parameter to the SearchLight in the last PR:  currently, we encounter an error if `scoring='roc_auc'` but `y` is not (0, 1):

``` python
import numpy as np
from mne.decoding.search_light import _SearchLight
from sklearn.linear_model import LogisticRegression

sl = _SearchLight(LogisticRegression(), scoring='roc_auc')
X = np.random.rand(100, 20, 2)
y = np.random.randint(0, 2, 100)
sl.fit(X, y)
sl.score(X, y)  # works

y = np.random.randint(2, 4, 100)
sl.fit(X, y)
sl.score(X, y)  # crashes
```

I suggest to add a sklearn.preprocessing.LabelEncoder() at fitting and use it before scoring. @agramfort do you think it's the right approach?

### PR DESCRIPTION
**Pull request by kaichogami**
Closes #3697
@kingjr

### CODE REVIEW

File: mne/decoding/search_light.py [hunk 1]
```diff
@@ -81,6 +81,12 @@ def fit(self, X, y):
         self : object
             Return self.
         """
+        if len(np.unique(y)) <= 2:
+            from sklearn.preprocessing import LabelEncoder
+             
+            le = LabelEncoder()
+            y = le.fit_transform(y)
```
Comment (kingjr) [inline mne/decoding/search_light.py]: don't do this at fitting, it might confuse the users if they passed y=[1, 2] but didn't use roc.

Instead, apply it at scoring. If len(np.unique(y_true)) == 2, then roc_auc_score(le.fit_transform(y_true))

File: mne/decoding/search_light.py [hunk 2]
```diff
@@ -228,7 +228,16 @@ def score(self, X, y):
...
+
+        if len(np.unique(y)) <= 2:
+            from sklearn.preprocessing import LabelEncoder
+                                     
+            le = LabelEncoder()
+            y = le.fit_transform(y)
+
         self._check_Xy(X)
+        le = LabelEncoder()
+        y = le.fit_transform(y)
```
Comment (kaichogami) [inline mne/decoding/search_light.py]: @kingjr I think the current change satisfies the behavior you have mentioned. Labels are only converted to `[0, 1]` while using score function. I have now removed this redundant `LabelEncoder`.

**Comment by kingjr** [comment]
ok can you add a test to ensure this?

**Comment by mmagnuski** [comment]
now only pep8 complains:
`mne/decoding/search_light.py:234:1: W293 blank line contains whitespace`

**Comment by kaichogami** [comment]
This should work now most likely.

File: mne/decoding/tests/test_search_light.py [hunk 3]
```diff
@@ -113,6 +120,13 @@ def transform(self, X):
         assert_array_equal(features_shape, [3, 4])
     assert_array_equal(y_preds[0], y_preds[1])
 
+    # Test when 'roc_auc' for binary labels
+    X = np.random.rand(5, 20, 2)
+    y = [4, 4, 3, 3, 3]
+    sl = _SearchLight(LogisticRegression(), scoring='roc_auc')
+    sl.fit(X, y)
+    sl.score(X, y)
```
Comment (kingjr) [inline mne/decoding/tests/test_search_light.py]: I think there's still a problem. If you fit with y in [4, 3], and score (without an roc_auc), score will change y_true in [4, 3] into y_true in [0, 1], but the estimator predictions will in [4, 3] won't they?

You need to add a test where X is not fully random but has some information, and check that roc_auc or accuracy give higher than chance scores.

Comment (kaichogami) [inline mne/decoding/tests/test_search_light.py]: Isn't that the desired behavior, to have the predictions in `y_true`?

**Comment by kingjr** [comment]
I believe that this has to be solved at sklearn level: https://github.com/scikit-learn/scikit-learn/pull/6874
